src/Catalyst.Node.POA.CE.Tests/IntegrationTests/PoaTestNode.cs

```diff
@@ -0,0 +1,177 @@
+                    Constants.SerilogJsonConfigFile
+                }
+               .Select(f => Path.Combine(Constants.ConfigSubFolder, f)), parentTestFileSystem, output);
+
+            Program.RegisterNodeDependencies(_containerProvider.ContainerBuilder,
+                excludedModules: new List<Type> {typeof(ApiModule), typeof(RpcServerModule)}
+            );
+            _containerProvider.ConfigureContainerBuilder(true, true);
+            OverrideContainerBuilderRegistrations();
+
+            _scope = _containerProvider.Container.BeginLifetimeScope(Name);
+            _node = _scope.Resolve<ICatalystNode>();
+
+            var keyStore = _scope.Resolve<IKeyStore>();
+            var keyRegistry = _scope.Resolve<IKeyRegistry>();
+            keyRegistry.AddItemToRegistry(KeyRegistryTypes.DefaultKey, privateKey);
+
+            keyStore.KeyStoreEncryptAsync(privateKey, nodeSettings.NetworkType, KeyRegistryTypes.DefaultKey)
+               .ConfigureAwait(false).GetAwaiter()
+               .GetResult();
+        }
+
+        public string Name { get; }
+
+        public IConsensus Consensus => _node.Consensus;
+
+        public async Task RunAsync(CancellationToken cancellationSourceToken)
+        {
+            await _node.RunAsync(cancellationSourceToken).ConfigureAwait(false);
+        }
+
+        public async Task StartSocketsAsync() { await _node.StartSocketsAsync(); }
+
+        public void Dispose() { Dispose(true); }
+
+        protected void OverrideContainerBuilderRegistrations()
+        {
+            _containerProvider.ContainerBuilder.RegisterInstance(new TestPasswordReader()).As<IPasswordReader>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_nodeSettings).As<IPeerSettings>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_rpcSettings).As<IRpcServerSettings>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_nodePeerId).As<PeerId>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_dfs).As<IDfs>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_memPool).As<IMempool<TransactionBroadcastDao>>();
+            _containerProvider.ContainerBuilder.RegisterInstance(_peerRepository).As<IPeerRepository>();
+            _containerProvider.ContainerBuilder.RegisterType<TestFileSystem>().As<IFileSystem>()
+               .WithParameter("rootPath", _nodeDirectory.FullName);
+            _containerProvider.ContainerBuilder.RegisterInstance(Substitute.For<IPeerDiscovery>()).As<IPeerDiscovery>();
+            var keySigner = Substitute.For<IKeySigner>();
+            keySigner.Verify(Arg.Any<ISignature>(), Arg.Any<byte[]>(), default).ReturnsForAnyArgs(true);
+            keySigner.CryptoContext.SignatureLength.Returns(64);
+            _containerProvider.ContainerBuilder.RegisterInstance(keySigner).As<IKeySigner>();
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposing) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)